### PR TITLE
OCLOMRS-61: Implement the 404 page

### PR DESCRIPTION
**Pull request by hadijahkyampeire**
# JIRA TICKET NAME:
[OCLOMRS-61: Implement the 404 page](https://issues.openmrs.org/browse/OCLOMRS-61)

# Summary:
This will always be displayed when the users try to reach a page that is not found

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -157,5 +157,25 @@ p{
...
+  position: relative;
+}
+.btn-not-found{
+  background-color: #495056;
+  border-color: white;
+  margin-bottom: -287%!important;
+}
+.error-template{
+  margin-top: 10px!important;
+}
```
Comment (shakirandagire) [inline src/App.css]: Please add the newline at the end of the file

File: src/App.js [hunk 2]
```diff
@@ -13,22 +13,55 @@ import DictionaryDisplay from './components/dashboard/container/DictionariesDisp
...
           <Route exact path="/" component={Authenticate(Login)} />
-          <Route exact path="/dashboard" component={Authenticate(SourceSearch)} />
-          <Route exact path="/dashboard/concepts" component={Authenticate(ConceptSearch)} />
-          <Route exact path="/dashboard/dictionaries" component={Authenticate(DictionaryDisplay)} />
-          <Route exact path="/dashboard/concepts/:organization/:name" component={Authenticate(SpecificConcept)} />
+          <Route
+            exact
+            path="/dashboard"
+            component={Authenticate(SourceSearch)}
+          />
```
Comment (Efosaok) [inline src/App.js]: please remove the arbitrary spaces to enable better code readability.

Comment (hadijahkyampeire) [reply src/App.js]: Thanks though locally its fine with no space I wonder how it comes, let me look into it

File: src/App.css (hunk 1 above)
Comment (hadijahkyampeire) [reply src/App.css]: Done, thanks for the reminder

**Comment by Efosaok** [comment]
LGTM

File: src/App.js [hunk 3]
```diff
@@ -13,6 +13,22 @@ import DictionaryDisplay from './components/dashboard/container/DictionariesDisp
...
+        <a href="/" className="btn btn-primary btn-lg btn-not-found">
+          <span className="fa fa-home" />
+            Take me back home{' '}
+        </a>
+      </div>
+
+    </div>
+  </div>
+);
+
```
Comment (emasys) [inline src/App.js]: This should be in a separate file

File: src/tests/Auth/notFound.test.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import { mount } from 'enzyme';
+import { NotFound } from '../../App';
+
+describe('Not found Component', () => {
+  it('should render without crashing', () => {
+    expect(mount.bind(null, <NotFound />)).not.toThrow();
+  });
+});
```
Comment (emasys) [inline src/tests/Auth/notFound.test.js]: You could take a snapshot to be sure the component is properly rendered.

Comment (hadijahkyampeire) [reply src/tests/Auth/notFound.test.js]: Ok thanks

File: src/App.js (hunk 3 above)
Comment (hadijahkyampeire) [reply src/App.js]: cool

**Pull request merged**